src/custom-components/ColorPickerModal.tsx

```diff
@@ -0,0 +1,163 @@
+import { ModalRoot, SliderField } from "../deck-components";
+import { useState, VFC } from "react";
+
+interface ColorPickerModalProps {
+  closeModal: () => void;
+  onConfirm?(HSLString: string): any;
+  title?: string;
+  defaultH?: number;
+  defaultS?: number;
+  defaultL?: number;
+  defaultA?: number;
+}
+
+export const ColorPickerModal: VFC<ColorPickerModalProps> = ({
+  closeModal,
+  onConfirm = () => {},
+  title = "Color Picker",
+  defaultH = 0,
+  defaultS = 100,
+  defaultL = 50,
+  defaultA = 1,
+}) => {
+  const [H, setH] = useState<number>(defaultH);
+  const [S, setS] = useState<number>(defaultS);
+  const [L, setL] = useState<number>(defaultL);
+  const [A, setA] = useState<number>(defaultA);
+
+  return (
+      <ModalRoot
+        bAllowFullSize
+        onCancel={closeModal}
+        onOK={() => {
+          onConfirm(`hsla(${H}, ${S}%, ${L}%, ${A})`);
+          closeModal();
+        }}
+      >
+        <style>
+        {`
+        :root {
+          --decky-color-picker-hvalue: ${H};
+          --decky-color-picker-svalue: ${S}%;
+          --decky-color-picker-lvalue: ${L}%;
+          --decky-color-picker-avalue: ${A};
+        }
+
+        /* This removes the cyan track color that is behind the slider head */
+        .ColorPicker_Container .gamepadslider_SliderTrack_Mq25N {
```

Comment:
Please switch to grabbing minified classes from their webpack modules to prevent breakage when updates occur. These modules should be added to static-classes.ts

Reply:
My newest commit added the required classes to static-classes.ts and references them in the component